index.js

```diff
@@ -101,14 +101,21 @@ function csurf (options) {
       return token
     }
 
+    var submitted = value(req)
```

Comment:
I would continue the refactoring around this in order to not call the `value` function if the method is ignored, similar to the current behavior.